### PR TITLE
Removed `urijs` and used `URL`

### LINKED ISSUE #952: Remove urijs and use `URL`
We use urijs for URIs in the browser, but modern browsers seem to have a `URL` type already built in.

Unless there's some reason not to, we should likely switch to using that

### PR DESCRIPTION
**Pull request by Swarnendu0123**
# In Progress: Fixes #952 
Removed `urijs` and used `URL`

----
📚 Documentation preview 📚: https://bowtie-json-schema--958.org.readthedocs.build/en/958/

### CODE REVIEW

**Comment by Swarnendu0123** [comment]
I see some conflicts are there, let me resolve them!

**Comment by Julian** [comment]
Possibly, but your CI also doesn't pass as-is.

**Comment by Swarnendu0123** [comment]
> Possibly, but your CI also doesn't pass as-is.

I think it's because I have changed the version     `"@typescript-eslint/parser": "^7.0.0",`

File: frontend/src/data/Dialect.ts [hunk 1]
```diff
@@ -32,9 +27,9 @@ export default class Dialect {
     this.firstPublicationDate = firstPublicationDate;
   }
 
-  async fetchReport(baseURI: URI = siteURI) {
-    const url = baseURI.clone().filename(this.path).suffix("json").href();
-    const response = await fetch(url);
+  async fetchReport(baseURL: URL = siteURL) {
+    const url = new URL(this.path + ".json", baseURL.href);
```
Comment (harrel56) [inline frontend/src/data/Dialect.ts:31]: I believe this behaves differently in some cases:
- old way: `https://harrel56.github.io/bowtie` resolves to  `https://harrel56.github.io/bowtie/draftX`
- new way:  `https://harrel56.github.io/bowtie` resolves to  `https://harrel56.github.io/draftX` - which is just wrong

Possibly there are differences in other parts too. Certainly, before merging it should be checked if GH pages on forks works correctly

**Comment by Julian** [comment]
Going to close this as I suspect it's equally easy for someone picking this up to start from scratch than to reference this, but feel free to send a new one if you get back to this.

(Please only do so after you get your other PR in mergeable state.)

**Pull request closed without merging**